Minor brick breaker cleanup & fixed wall generation

dan-rusu-inc opened the pull request.

daniel-rusu committed:
Minor brick breaker cleanup & fixed wall generation

daniel-rusu committed:
Delete Wall, BoxEntity, and CircleEntity

Delete Wall, BoxEntity, and CircleEntity since they are redundant.

davidrusu commented on src/gameengine/entities/Entity.java:
```diff
@@ -10,7 +10,7 @@
 import java.awt.*;
 
-public abstract class Entity {
+public class Entity {
```
👍 yay!

less `abstract` is better :)

daniel-rusu committed:
Improve code re-use by separating behaviors by using default methods in Interfaces

This is an idea for how we can improve code re-use.

One of the things that ECS systems fixes is that you end up with
multiple classes that have to re-implement common code because they
cannot share the same parents in the hierarchy.  Java 8 introduced
default methods in Interfaces so we can take advantage of that.

When defining a new class, we would simply define the new class as
implementing all the interfaces with the default behaviors that we want
(without needing to re-implement those behaviors).  So if a class
implements DrawShape then its shape is automatically drawn.

davidrusu reviewed (approved):
:/ the class hierarchy will just turn into a interface hierarchy, and the interfaces now have implicit state because that EntityCapabilities interface is directly tied to the entity class and exposes most of the state to the interface.

You can merge this but I feel it will just make things complicated with not much benefit.

I think it would be better to just have a static class of utilities and make the user call w/e methods they need.

ghost commented:
Leaving changes in fork
